src/core/mac/mac.cpp

```diff
@@ -577,27 +579,53 @@ otError Mac::ReleaseRadioChannel(void)
 
 otError Mac::SetNetworkName(const char *aNetworkName)
 {
-    otError error = OT_ERROR_NONE;
+    return SetNetworkName(aNetworkName, static_cast<uint8_t>(strnlen(aNetworkName, OT_NETWORK_NAME_MAX_SIZE + 1)));
+}
+
+otError Mac::SetNetworkName(const char *aBuffer, uint8_t aLength)
+{
+    otError       error = OT_ERROR_NONE;
+    otNetworkName newName;
+
+    while ((aLength > 0) && (aBuffer[aLength - 1] == 0))
+    {
+        aLength--;
+    }
+
+    VerifyOrExit(aLength <= OT_NETWORK_NAME_MAX_SIZE, error = OT_ERROR_INVALID_ARGS);
 
-    VerifyOrExit(strlen(aNetworkName) <= OT_NETWORK_NAME_MAX_SIZE, error = OT_ERROR_INVALID_ARGS);
+    memcpy(newName.m8, aBuffer, aLength);
```

Comment:
How about doing the `memcmp` then `memcpy`, rather than `memcpy`, `strcmp`, `strlcpy`?  That removes the need for an additional buffer and copy.

Reply:
:thumbsup: